packages/core-ethereum/src/indexer/index.ts

```diff
@@ -291,44 +291,67 @@ class Indexer extends EventEmitter {
    */
   private async processPastEvents(fromBlock: number, maxToBlock: number, maxBlockRange: number): Promise<number> {
     const start = fromBlock
-    let failedCount = 0
 
-    while (fromBlock < maxToBlock) {
-      const blockRange = failedCount > 0 ? Math.floor(maxBlockRange / 4 ** failedCount) : maxBlockRange
-      // should never be above maxToBlock
-      let toBlock = Math.min(fromBlock + blockRange, maxToBlock)
+    return new Promise<number>((resolve) => {
```

Comment:
Same as before, too much inlining and missing documentation. Introduction of a worker type which is also undocumented.